api/src/main/java/org/apache/iceberg/types/Comparators.java

```diff
@@ -104,7 +104,10 @@ private Comparators() {}
 
     private StructLikeComparator(Types.StructType struct) {
       this.comparators = struct.fields().stream()
-          .map(field -> internal(field.type()))
+          .map(field -> field.isOptional() ?
```

Comment:
Good catch.

Reply:
I should add this to the list comparator as well.